integration_tests/src/main/python/conditionals_test.py

```diff
@@ -67,11 +69,15 @@ def test_if_else_map(data_gen):
 
 @pytest.mark.order(1) # at the head of xdist worker queue if pytest-order is installed
 @pytest.mark.parametrize('data_gen', all_gens + all_nested_gens + decimal_128_gens, ids=idfn)
-def test_case_when(data_gen):
+@pytest.mark.parametrize('const_bool', [True, False, None])
+def test_case_when(data_gen, const_bool):
     num_cmps = 20
     s1 = gen_scalar(data_gen, force_no_nulls=not isinstance(data_gen, NullGen))
-    # we want lots of false
-    bool_gen = BooleanGen().with_special_case(False, weight=1000.0)
+    if const_bool:
```

Comment:
Same issue where `const_bool == False` will not go through this code path.  `if const_bool is not None` seems more appropriate here.

Reply:
Nice catch, updated.